### DISCUSSION TITLE
AttributeError: partially initialized module 'xgboost' has no attribute 'DMatrix' (most likely due to a circular import)

### DISCUSSION THREAD

**Discussion by EvanGertis** [Trusted-AI/adversarial-robustness-toolbox#1722] [Q&A] [answered]
I am running into an issue using the xgboost classifer from adversarial robustness toolbox (ART).

AttributeError: partially initialized module 'xgboost' has no attribute 'DMatrix' (most likely due to a circular import)

clone the ART repo.
cd into the examples dir.
run the xgboost example
Expected behavior The program should print out the accuracy for the classification of the MNIST Model.

Actual Traceback (most recent call last): File "/Users/evangertis/development/PythonAutomation/ART_examples/xgboost.py", line 7, in import xgboost as xgb File "/Users/evangertis/development/PythonAutomation/ART_examples/xgboost.py", line 31, in dtrain = xgb.DMatrix(x_train, label=np.argmax(y_train, axis=1)) AttributeError: partially initialized module 'xgboost' has no attribute 'DMatrix' (most likely due to a circular import)

Screenshots

System information (please complete the following information):

MacOS
Python 3.9
Latest commit 3ff566f
XGBoost: 1.5.2

**Comment by beat-buesser** [comment, marked as answer]
This questions seems to be a duplicate of #1717. Let's continue the discussion in #1717.
